### PR TITLE
[oss-fuzz integration]: misc fixes in fuzzing tests

### PR DESCRIPTION
**Pull request by RiSKeD**
These changes are intended as fixes to the fuzzing tests when running in the oss-fuzz environment.
1. Reduce input size of one test so libfuzzer does not run out of memory 
2. Use a temporary directory per fuzzing input run so no execution is influenced by previous runs
Signed-off-by: Fabian Wienand <fabian.wienand@9elements.com>

### CODE REVIEW

File: pkg/boot/grub/config_test.go [hunk 1]
```diff
@@ -159,11 +139,37 @@ func FuzzParseGrubConfig(f *testing.F) {
...
+
+		// do not allow arbitrary files reads
+		if bytes.Contains(data, []byte("include")) {
 			return
 		}
 
-		err := os.WriteFile(path, data, 0o777)
+		baseDir := t.TempDir()
+
+		dirPath := baseDir + "/EFI/uefi"
```
Comment (rminnich) [inline pkg/boot/grub/config_test.go:153]: might as well fix this now: filepath.Join, not +
should we ever wish to run on windows, it ought to be, possibly:
filepath.Join(baseDir, "EFI", "uefi")

File: pkg/boot/syslinux/syslinux_test.go [hunk 2]
```diff
@@ -733,10 +731,19 @@ func FuzzParseSyslinuxConfig(f *testing.F) {
...
+		if bytes.Contains(data, []byte("include")) {
 			return
 		}
 
+		dirPath := t.TempDir()
+
+		path := filepath.Join(dirPath, "isolinux.cfg")
 		err := os.WriteFile(path, data, 0o777)
 		if err != nil {
 			t.Errorf("Failed to create configfile '%v':%v", path, err)
```
Comment (rminnich) [inline pkg/boot/syslinux/syslinux_test.go:749]: Shouldn't this be Fatalf? If the write fails, can the test really continue?
or even t.Errorf followed by continue in the if block?

File: pkg/boot/grub/config_test.go (hunk 1 above)
Comment (RiSKeD) [reply pkg/boot/grub/config_test.go]: fixed with ee962ad

File: pkg/boot/syslinux/syslinux_test.go (hunk 2 above)
Comment (RiSKeD) [reply pkg/boot/syslinux/syslinux_test.go]: Same as the other instance, fixed with the newest commit

**Review by rminnich — APPROVED** [review]

**Review by rminnich — APPROVED** [review]

**Pull request merged**